src/main/java/org/opensearch/neuralsearch/search/collector/HybridTopFieldDocSortCollector.java

```diff
@@ -0,0 +1,410 @@
+                return (HybridQueryScorer) scorer;
+            }
+            for (Scorable.ChildScorable childScorable : scorer.getChildren()) {
+                HybridQueryScorer hybridQueryScorer = getHybridQueryScorer(childScorable.child);
+                if (Objects.nonNull(hybridQueryScorer)) {
+                    log.debug(
+                        String.format(
+                            Locale.ROOT,
+                            "found hybrid query scorer, it's child of scorer %s",
+                            childScorable.child.getClass().getSimpleName()
+                        )
+                    );
+                    return hybridQueryScorer;
+                }
+            }
+            return null;
+        }
+
+        /*
+        Increment total hit count and validate if threshold is reached.
+         */
+        protected void incrementTotalHitCount() throws IOException {
+            totalHits++;
+            hitsThresholdChecker.incrementHitCount();
+            if (scoreMode().isExhaustive() == false
+                && getTotalHitsRelation() == TotalHits.Relation.EQUAL_TO
+                && hitsThresholdChecker.isThresholdReached()) {
+                // for the first time hitsThreshold is reached, notify all comparators about this
+                for (LeafFieldComparator comparator : comparators) {
+                    comparator.setHitsThresholdReached();
+                }
+                setTotalHitsRelation(TotalHits.Relation.GREATER_THAN_OR_EQUAL_TO);
+            }
+        }
+
+        /*
+        Collect hit and add the value of the sort field in the comparator.
+         */
+        protected void collectHit(int doc, int hitsCollected, int subQueryNumber, float score) throws IOException {
+            // Startup transient: queue hasn't gathered numHits yet
+            int slot = hitsCollected - 1;
+            // Copy hit into queue
+            if (numHits > 0) {
+                comparators[subQueryNumber].copy(slot, doc);
+                add(slot, doc, compoundScores[subQueryNumber], subQueryNumber, score);
+                if (queueFull[subQueryNumber]) {
+                    comparators[subQueryNumber].setBottom(bottom.slot);
+                }
+            } else {
+                queueFull[subQueryNumber] = true;
+            }
+        }
+
+        /*
+         * This hit is competitive - replace bottom element in queue & adjustTop
+         */
+        protected void collectCompetitiveHit(int doc, int subQueryNumber) throws IOException {
+            // This hit is competitive - replace bottom element in queue & adjustTop
+            if (numHits > 0) {
+                comparators[subQueryNumber].copy(bottom.slot, doc);
+                updateBottom(doc, compoundScores[subQueryNumber]);
+                comparators[subQueryNumber].setBottom(bottom.slot);
+            }
+        }
+
+        protected boolean thresholdCheck(int doc, int subQueryNumber) throws IOException {
```

Comment:
why protected?

Reply:
Because in SimpleFieldCollector and PagingFieldCollector **getLeafCollector** method, we initialize the object of HybridTopFieldDocLeafCollector  by defining collect method in it. Now as HybridTopFieldDocLeafCollector is an **abstract** class we need this access modifier to be **protected** so that the places where we initialize the abstract leaf collector we can call this thresholdcheck method in the collect method.